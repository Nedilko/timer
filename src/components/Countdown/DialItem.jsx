import PropTypes from "prop-types";
import { formatNumberTwoDigit } from "../../utils/helperFunctions";

DialItem.propTypes = {
  title: PropTypes.string.isRequired,
  numbers: PropTypes.number.isRequired,
  isLast: PropTypes.bool,
};

function DialItem({ title, numbers, isLast }) {
  return (
    <div
      className={`inline-flex flex-col w-1/4 px-1 ${
        isLast
          ? ""
          : "after:absolute after:text-2xl after:content-[':'] after:absolute after:right-0 after:bottom-0 relative"
      }`}
    >
      <div>{title}</div>
      <div className="text-2xl">{formatNumberTwoDigit(numbers)}</div>
    </div>
  );
}

export default DialItem;
