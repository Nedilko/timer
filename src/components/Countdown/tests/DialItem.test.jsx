import DialItem from "../DialItem"
import { render, screen } from "@testing-library/react"

describe('DialItem', () => { 
    it('should match snapshot',() => {
        render( <DialItem title="test" numbers={5} /> )

        expect(screen.getByRole("dialitem")).toMatchSnapshot();
    })

    it('should render string "05"', () =>{
        render( <DialItem title="test" numbers={5}/> )

        expect(screen.getByRole("dialitem")).toHaveTextContent('05');
    })

    it('should render title "test"',() => {
        render( <DialItem title="test" numbers={5}/> )

        expect(screen.getByRole("dialitem")).toHaveTextContent('test');
    })

    it('should have extra classes for isLast = false',() => {   
        render( <DialItem title="test" numbers={5}/> )

        expect(screen.getByRole("dialitem")).toHaveClass("after:absolute after:text-2xl after:content-[':'] after:absolute after:right-0 after:bottom-0 relative")
    })
    
    it('should not have extra classes for isLast = true',() => {
        render( <DialItem title="test" numbers={5} isLast={true}/> )

        expect(screen.getByRole("dialitem")).not.toHaveClass("after:absolute after:text-2xl after:content-[':'] after:absolute after:right-0 after:bottom-0 relative")        
    })
 })