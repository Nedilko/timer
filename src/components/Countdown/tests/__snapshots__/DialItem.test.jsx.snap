// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialItem should match snapshot 1`] = `
<div
  class="inline-flex flex-col w-1/4 px-1 after:absolute after:text-2xl after:content-[':'] after:absolute after:right-0 after:bottom-0 relative"
  role="dialitem"
>
  <div>
    test
  </div>
  <div
    class="text-2xl"
  >
    05
  </div>
</div>
`;
